wix/wix-style-react
# <Search/> + <InputWithOptions/> - Clear value in controlled mode

** Blocker for WSR 8 **
There's a bug that in a controlled `<Search/>` the value isn't really cleared when clicking the clear button.

How to reproduce:
1. Paste this code in the playground
```
class MyComponent extends React.Component {
  state={
    value: '',
  };

  render() {
    return (
    <Search
    updateControlledOnClear
      options={Array(26)
        .fill(0)
        .map((_, id) => ({
          id,
          value: `Option ${String.fromCharCode(97 + id)}`,
        }))}
      showOptionsIfEmptyInput={false}
      closeOnSelect={false}
      value={this.state.value}
      onChange={e => this.setState({value: e.target.value})}
      onSelect={option => this.setState({value: option.value})}
    />
    );
  }
}
```
2. Click/Focus on the search bar and type any letter, for example the letter `p`.
3. Click the clear button on the far right side of the component.

If you followed all steps correctly, the result should be that the value is cleared from the input, but the **dropdown is open** and the **letter `p` is still marked**.
Like this screenshot:
![image](https://user-images.githubusercontent.com/35173937/77701538-3d884980-6fbf-11ea-83a1-8952e45874ac.png)


The wanted result, which you can achieve by removing the prop `updateControlledOnClear`, is once clicked the clear button, the **dropdown is closed** and the **letter `p` is not marked**.
